script/src/main.rs

```diff
@@ -0,0 +1,61 @@
+
+use sqlx::PgPool;
+use sqlx::Result;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    dotenv::dotenv().expect("Unable to load environment variables from .env file");
+
+    let db_url = std::env::var("DATABASE_URL").expect("Unable to read DATABASE_URL env var");
+    let pool = PgPool::connect(&db_url).await?;
+
+    // Define a struct to represent the rows returned by the query
+    #[derive(sqlx::FromRow)]
+    struct JobId {
+        id: i32,
+    }
+
+    // Define the SQL query to identify job IDs where total_processed in email_results is equal to total_records in bulk_jobs
+    let job_ids_to_delete_query = r#"
+        SELECT b.id
+        FROM bulk_jobs b
+        JOIN (
+            SELECT job_id, COUNT(*) as total_processed
+            FROM email_results
+            GROUP BY job_id
+        ) e ON b.id = e.job_id
+        WHERE b.total_records = e.total_processed
+    "#;
+
+    // Fetch the list of job IDs that match the criteria
+    let job_ids_to_delete: Vec<JobId> = sqlx::query_as(job_ids_to_delete_query)
+        .fetch_all(&pool)
+        .await?;
+
+    for job_id in job_ids_to_delete {
+        // Before deleting from bulk_jobs, delete the corresponding records from email_results
+        let delete_email_results_query = "DELETE FROM email_results WHERE job_id = $1";
```

Comment:
It's generally a better idea to push down logic into SQL than loop fire queries. Consider using `WHERE job_id IN (id1, id2 .....)` [so](https://stackoverflow.com/q/5803472).

Also if the email_result delete succeeds and the bulk email delete fails the db will be in an inconsistent state. It's better to use transactions to avoid that. Both deletes work or neither works. (link)[https://www.postgresql.org/docs/current/tutorial-transactions.html]

Reply:
Using the sqlx for creating and committing transactions is an interesting way to go about it.

Do you think this is better than the pure sql approach with "BEGIN ... END"?